### ISSUE TITLE
Deadlock in agent state processing

### ISSUE THREAD

**Issue by wouterdb** [inmanta/inmanta-core#4844]
This was observed on a test orchestrator

```
2022-09-16 03:53:22.259 UTC [25779] DETAIL:  Process 25779 waits for ShareLock on transaction 189497482; blocked by process 25459.
        Process 25459 waits for ShareLock on transaction 189497484; blocked by process 25779.
        Process 25779: DELETE FROM agent WHERE environment=$1
        Process 25459: SELECT  *  FROM agent WHERE environment=$1 AND name=$2 FOR UPDATE

```

This is probably the environment.clear conflicting with the agent state update

```
# rpm -qi inmanta-service-orchestrator
Name        : inmanta-service-orchestrator
Version     : 4.4.5
Release     : 0.dev20220904021800.el7
Architecture: x86_64
Install Date: Mon 05 Sep 2022 12:00:11 AM UTC
Group       : Development/Languages
Size        : 73886834
License     : EULA
Signature   : RSA/SHA256, Sun 04 Sep 2022 02:32:35 AM UTC, Key ID 6267cd49f0157b1b
Source RPM  : inmanta-service-orchestrator-4.4.5-0.dev20220904021800.el7.src.rpm
Build Date  : Sun 04 Sep 2022 02:32:14 AM UTC
Build Host  : 173035ae59d5450c9ae8fdd020987036
Relocations : (not relocatable)
URL         : http://inmanta.com
Summary     : Inmanta Service Orchestrator
Description :

```

**Comment by sanderr** [comment]
This may or may not have been solved with #4889

**Comment by bartv** [comment]
Lets close it then.
